src/Detector.php

```diff
@@ -67,19 +67,26 @@ public function determineMimeType(\SplFileObject $file): string
     /**
      * Read width and height from a image file.
      *
-     * @param string $filePath
+     * @param \SplFileObject $file File object
      * @return array [$width, $height]
      * @throws \InvalidArgumentException Unreadable SplFileInfo
      */
-    public function determineDimensions(string $filePath): array
+    public function determineDimensions(\SplFileObject $file): array
     {
-        $stream = new \SplFileInfo($filePath);
-        if (!$stream->isReadable()) {
-            throw new \InvalidArgumentException('File path ' . $filePath . ' is not existed');
+        if (false !== $path = $file->getRealPath()) {
+            $info = getimagesize($path);
+
+            return [$info[0], $info[1]];
+        }
+
+        $file->rewind();
```

Comment:
[`SplFileObject::rewind`](https://secure.php.net/manual/en/splfileobject.rewind.php) may throw a `RuntimeException`. This should be checked and handled somehow, or it must be documented that we may throw that to the end-user. Alternatively [`SplFileObject::fseek`](https://secure.php.net/manual/en/splfileobject.fseek.php) may be used, though someone should confirm that it doesn’t throw:

```php
$file->fseek(0);
```
```

Reply:
I prefer add the `RuntimeException` comment annotation to notice the end-user :+1:.